api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirObservationDaoImpl.java

```diff
@@ -175,4 +220,53 @@ protected Obs deproxyResult(Obs result) {
 		obs.setConcept(deproxyObject(obs.getConcept()));
 		return obs;
 	}
+	
+	int getMax(SearchParameterMap theParams) {
+		return ((NumberParam) theParams.getParameters(FhirConstants.MAX_SEARCH_HANDLER).get(0).getParam()).getValue()
+		        .intValue();
+		
+	}
+	
+	protected List<String> handleRanking(List<Object[]> list, int max) {
+		List<String> results = new ArrayList<>();
```

Comment:
Since, in this case, it seems like `results` can never be higher than `list.size()`, it might make sense to change this to:

```
List<String> results = new ArrayList<>(list.size());
```